File: 2220-minimum-bit-flips-to-convert-number/2220-minimum-bit-flips-to-convert-number.java
class Solution {
    public int minBitFlips(int start, int goal) {
        int t = start ^ goal, cnt = 0;
        
        while (t != 0) {
            if (t % 2 != 0) cnt++;
            t >>= 1;
        }
        
        return cnt;
    }
}

File: 2220-minimum-bit-flips-to-convert-number/2220-minimum-bit-flips-to-convert-number.cpp
class Solution {
public:
    int minBitFlips(int start, int goal) {
        int xr = start ^ goal;
        
        int res = 0;
        
        while (xr > 0) {
            xr = xr & (xr - 1);
            res++;
        }
        
        return res;
    }
};